pysal/contrib/geotable/utils.py

```diff
@@ -50,13 +49,40 @@ def to_gdf(df, geom_col='geometry', **kw):
     """
     from geopandas import GeoDataFrame
     from shapely.geometry import asShape as sShape
-    df[geom_col] = df[geom_col].apply(sShape)
-    return GeoDataFrame(df, geometry=geom_col, **kw)
+    out = df.copy()
+    out[geom_col] = out[geom_col].apply(sShape)
+    out = GeoDataFrame(out, geometry=geom_col, **kw)
+    return out
 
-def insert_metadata(df, obj, name=None, inplace=True, overwrite=False):
+def insert_metadata(df, obj, name=None, inplace=False, overwrite=False):
```

Comment:
This was to allow a user to insert and propagate things into their dataframe by modifying the `dataframe._metadata`. Stuff in metadata persists over copies. 

I added it to explore integrating the weights into the dataframe that way, so that you can monkeypatch a weights object into your dataframe and refer to it as `dataframe.W`.